modules/core/navigation/navigation.js

```diff
@@ -102,6 +102,16 @@ async function navigate(url, loaderMessage) {
         if (!html || typeof html !== 'string') {
             throw new Error("Failed navigating to page, empty or invalid response body.");
         }
+
+        // load custom javascript
+        const scripts = extractCustomScripts($(html));
+        loadCustomScripts(scripts);
```

Comment:
If we load the custom scripts before the main content is replaced, wouldn't some of them fail by actually working on the old page content?

Reply:
Seems like none of them actually depend on the DOM content; they are all state data. So, none would fail.